gems/sorbet-runtime/lib/types/types/typed_hash.rb

```diff
@@ -3,22 +3,38 @@
 
 module T::Types
   class TypedHash < TypedEnumerable
-    # Technically we don't need these, but they are a nice api
-    attr_reader :keys, :values
-
     def underlying_class
       Hash
     end
 
     def initialize(keys:, values:)
-      @keys = T::Utils.coerce(keys)
-      @values = T::Utils.coerce(values)
-      @type = T::Utils.coerce([keys, values])
+      @inner_keys = keys
+      @inner_values = values
+    end
+
+    # Technically we don't need this, but it is a nice api
+    def keys
+      @keys ||= T::Utils.coerce(@inner_keys)
+    end
+
+    # Technically we don't need this, but it is a nice api
+    def values
+      @values ||= T::Utils.coerce(@inner_values)
+    end
+
+    def type
+      @type ||= T::Utils.coerce([@inner_keys, @inner_values])
```

Comment:
Not blocking: I wonder if it's worth using `@keys, @values` here instead, because then the coerce will not have to do the work of coercing `@inner_keys` and `@inner_values` twice.

Reply:
Good point: cda3bbd 👍

Reply:
(I assumed you meant the `keys` and `values` methods rather than accessors)